docs/lib/datastore/fragments/ios/auth-model.md

```diff
@@ -0,0 +1,97 @@
+## Updated schema
+
+The following schema is an example of a model named `Todo` that is readable and writable by the creator, and read only to everyone else.
+
+```graphql
+type Todo
+    @model
+    @auth(rules: [
+        { allow: owner, ownerField: "owner", operations: [create, update, delete] },
+    ]) {
+    id: ID!
+    content: String!
+    owner: String
+}
+```
+
+The schema will allow users to query, list, and subscribe to model creation, updates, deletions by other users. A user can only update and delete their own model, that is a model can not be updated or deleted by another user.
+
+The `@auth` directive when used with Amplify DataStore supports the following fields:
+
+  - `allow` with "owner" and "groups"
+  - `ownerField`
+  - `groups`
+  - `operations`
+
+## Configure API with Amazon Cognito user pool
+
+While using the `@auth` directive in your schema to limit read/write access, we require authentication to be integrated into your app in order to differentiate who the owner is. To set this up issue the command:
+
+```console
+amplify update api
+```
+
+then follow the prompts:
+
+```console
+? Please select from one of the below mentioned services: 
+  `GraphQL`
+? Select from the options below 
+  `Update auth settings`
+? Choose the default authorization type for the API 
+  `Amazon Cognito User Pool`
+```
+
+Once finished, run `amplify push` to provision your services in the backend.
+
+## Configuration
+
+Upon successfully running `amplify push` per the last step, `amplifyconfiguration.json` will have a section under "awsAPIPlugins" and "awsCoognitoAuthPlugin" with the corresponding API and Auth configurations. Your `amplifyconfiguration.json` may look like this:
+
+```json
+{
+    "api": {
+        "plugins": {
+            "awsAPIPlugin": {
+                // ...
+            }
+        }
+    },
+    "auth": {
+        "plugins": {
+            "awsCognitoAuthPlugin": {
+                // ...
+            }
+        }
+    }
+```
+
+Add "AWSCognitoAuthPlugin" and "AWSAPIPlugin" to Amplify before configuring
+
+```swift
+Amplify.add(plugin: AWSCognitoAuthPlugin())
+Amplify.add(plugin: AWSAPIPlugin())
+Amplify.configure()
+```
+
+## Authenticate before Save
```

Comment:
we need a better title for this because we probably want to authenticate before any operation that has fine grain controls(save, delete, update, etc.)